Formula/cnosdb.rb

```diff
@@ -0,0 +1,25 @@
+class Cnosdb < Formula
+  desc "Open source distributed time series database"
+  homepage "https://www.cnosdb.com"
+  url "https://github.com/cnosdb/cnosdb/archive/refs/tags/v1.0.2.tar.gz"
+  sha256 "ae412d0944b64c9b39dc1edc66f7b6f712b85bc5afad354c12b135ae71017100"
+  license "MIT"
+  head "https://github.com/cnosdb/cnosdb.git"
+
+  livecheck do
+    url "https://github.com/cnosdb/cnosdb/releases/latest"
+    regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOBIN"] = buildpath
+    system "go", "install", "./..."
+    bin.install %w[cnosdb-cli cnosdb-ctl cnosdb-meta cnosdb-inspect cnosdb-tools]
+  end
+
+  test do
+    system "#{bin}/cnosdb-cli"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thank you. I'll figure out the solution to it.